File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
from math import factorial

def solution(n):
    result = 0
    for i in range(n // 2 + 1):
        cnt_one = n - 2 * i
        cnt_total = cnt_one + i
        result += (factorial(cnt_total) // (factorial(cnt_one) * factorial(i))) % 1234567
    return result % 1234567
        
    

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.js
// 피보나치 문제였다 ... 오메 ...
function solution(n) {
    let a = 1, b = 2;
    for(let i = 2 ; i < n ; i++){
        [a, b] = [b, (a+b) % 1234567];
    }
    return n === 1 ? a : b;
}